powerline.el

```diff
@@ -130,9 +130,22 @@ This is needed to make sure that text is properly aligned."
 ;;
 ;; see https://github.com/milkypostman/powerline/issues/58
 ;;
-(defun powerline-delete-cache ()
-  (set-frame-parameter nil 'powerline-cache nil))
-(add-hook 'desktop-save-hook 'powerline-delete-cache)
+;; It is better to put the following code into your init file.
+;; (push '(powerline-cache . :never) frameset-filter-alist)
+;;
+(defun powerline-delete-cache (&optional frame)
+  "Set the FRAME cache to nil."
+  (set-frame-parameter frame 'powerline-cache nil))
+
+(defvar frameset-filter-alist)
```

Comment:
I added `(defvar frameset-filter-alist)` for suppressing a warning message from byte compilation because `frameset-filter-alist` is defined in frameset.el introduced in Emacs 24.4.